anotha reactive teleport armor nerf

PKPenguin321 opened the pull request:
https://www.youtube.com/watch?v=FSswEzWJdWo

it now has a cooldown between teleports that is also applied when it gets EMPed

:cl: PKPenguin321
tweak: The RD's reactive teleport armor now has a cooldown between teleports. EMPing it will cause it to shut down and have a longer cooldown applied.
/:cl:

PKPenguin321 committed:
reactive teleport armor nerf

it now shuts off automatically whenever it procs a teleport

PKPenguin321 commented:
oh and uh this will probably conflict with https://github.com/tgstation/-tg-station/pull/16207

kevinz000 commented:
One or the other fam

PKPenguin321 commented:
Or both familia

CPTANT commented:
This is a nerf by making the interface shit. Nothing will change for people who push the hotkey every time they are hit.

PKPenguin321 commented:
1. There is no hotkey for action buttons
2. If you are in the middle of intense combat there is a good chance that your first thought won't always be on reactivating your armor

Lumipharon commented:
I still think just making it battery powered would be easiest - would also let emp's work on it as a hard counter.

PKPenguin321 commented:
apparently EMPs already do turn it off, but it can be immediately turned back on with 0 cooldown

Lumipharon commented:
well that's a bit dumb - you can't reset headsets instantly for example.

PKPenguin321 commented:
tru, maybe i can make that another nerf for another pr

@Fox-McCloud i literally cannot fathom why you would -1 this, i thought you of all people would support this at least

CPTANT commented:
Still nerf by shit interface.

If you don't want it to constantly activate give it a cooldown time before it re-activates itself after each teleport.

PKPenguin321 commented:
>Still nerf by shit interface.

i disagree, having it shut down every time is a mechanical function, not an interface one. nerf by shit interface would be something like making the action button way tiny.

Lumipharon commented:
it is an interface nerf - there is no cooldown on toggling the vest on or off, the only limit is how quickly you can move your mouse over to press the button.

For example, if I had a hotkey to toggle the vest, then this nerf would essentially not effect me - it's the interface which makes this inconvenient.

Really though, battery powered, emp disables it temporarily (if it doesn't drain the whole thing), and a cooldown on how often it could operate and it would be 100% fine even without the gay radiation thing.

PKPenguin321 committed:
there is now a hearty delay between each teleport

and a delay that is twice as long when the armor gets EMPed

PKPenguin321 commented:
>a cooldown on how often it could operate and it would be 100% fine

good idea i made that in the latest commit

ChangelingRain commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -184,9 +187,25 @@
+		active = 0
+		src.icon_state = "reactiveoff"
+		src.item_state = "reactiveoff"
+		cooldown++
```
Instead of increasing the cooldown var, set it to `world.time + 100` and check if `world.time < cooldown` on line 140.
This basically replaces lines 193-197 with one line.
Also you forgot to define the cooldown var

PKPenguin321 replied:
cooldown var is actually defined under /obj/item/clothing

ChangelingRain replied:
in that case make sure nothing else sets it, or you might have issues

ChangelingRain commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -184,9 +187,25 @@
 	return 0
 
+/obj/item/clothing/suit/armor/reactive/teleport/emp_act(severity)
+	cooldown++
```
Same thing here.

PKPenguin321 committed:
makes cooldown use a new var and cleaner code

thanks @changelingrain

KorPhaeron commented:
Why not just add a cooldown between teleports instead of making people mash on the button until the cooldown is done?

Lumipharon commented:
^

phil235 commented on code/modules/clothing/suits/armor.dm:
```diff
@@ -127,6 +127,7 @@
 	name = "reactive armor"
 	desc = "Doesn't seem to do much for some reason."
 	var/active = 0
+	var/cd = 0//cooldown specific to reactive armor
```
shitty var name

PKPenguin321 replied:
"cooldown" is taken, cd is just an acronym. What would you have me make it?

phil235 replied:
"reactive_armor_cooldown"

phil235 replied:
Are you sure you can't use obj/item/clothing/var/cooldown? (added in https://github.com/tgstation/-tg-station/commit/39aaea6f283be376c47b03046c2d901ba0922713#diff-c0790df45ca6836b3267c5b4f75ad285R15)

PKPenguin321 replied:
I did originally but changelingrain said to change it reeeew

PKPenguin321 commented:
>Why not just add a cooldown between teleports instead of making people mash on the button until the cooldown is done?

will do

PKPenguin321 committed:
removes the need to reenable telearmor after the cooldown ends

you still need to reenable it manually if you get EMPed, though, as it has always been

PKPenguin321 commented:
should be good to go now @KorPhaeron

Lumipharon commented:
:+1: Honestly with this change, providing the CD is long enough, the shitty radiation side effect doesn't even need to be there.

duncathan commented:
Code is fine

PKPenguin321 commented:
:ok_hand:
